### DISCUSSION TITLE
FAN Controller

### DISCUSSION THREAD

**Discussion by Jigsaw0815** [RavenSystem/esp-homekit-devices#2247] [Q&A]
Hi, I'm new here and am currently trying to integrate my fan into Homekit.
Unfortunately, I can't get it to work. The fan always turns at full speed. The lamp works as intended.
y    4
w   5
1  13
2  12
3  14
ir  16

{
  "c":{"io":[[[4,5,13,12,14],2],[[4,5],7,0,0,2]],"l":13},
  "a":[{
    "t":65,
    "0":{"r":[[13],[12],[14]]},
    "y0":[
      {"v": 1, "0":{"r":[[13,1,0.5],[12],[14,0,0.6]]} },
      {"v":33, "0":{"r":[[13,1,0.5],[12,0,0.6],[14,0,0.6]]} },
      {"v":67, "0":{"r":[[13,1],[12],[14]]} }
    ],
    "es":[{
        "t":30,
	    "g":[4,5],
	    "s":1,
	"mp":1
    }]
    }
  ]
}

**Comment by RavenSystem** [comment]
{
"c":{"io":[[[13,12,14],2],[[4],7],[[5],7,0,0,2]]},
"a":[{
"t":65,
"0":{"r":[[13],[12],[14]]},
"y0":[
{"v": 1, "0":{"r":[[13,1],[12],[14]]} },
{"v":33, "0":{"r":[[13],[12,1],[14]]} },
{"v":67, "0":{"r":[[13],[12],[14,1]]} }
],
"es":[{
"t":30,
"g":[4,5],
"s":1,
"mp":1
}]
}
]
}

**Reply by Jigsaw0815** [reply]
Thank You, Can you explane what i did wron ? i see the different in the code. Have to try if speed change but for now it looks Good.

**Comment by Jigsaw0815** [comment]
Speed Change not work did you have an idea ?
